src/java.base/windows/native/libnio/ch/Net.c

```diff
@@ -239,6 +247,14 @@ Java_sun_nio_ch_Net_connect0(JNIEnv *env, jclass clazz, jboolean preferIPv6, job
         NET_EnableFastTcpLoopbackConnect((jint)s);
     }
 
+    if (AF_INET == sa.sa4.sin_family) {
+        char buf[INET_ADDRSTRLEN];
+        Trc_nio_ch_Net_connect4((jlong)s, inet_ntop(AF_INET, &sa.sa4.sin_addr, buf, sizeof(buf)), port, sa_len);
```

Comment:
Failure of `inet_ntop` should be handled here (and on line 255).

Reply:
I think it's fine the way it is, unless you have another suggestion. I think it's better to print the tracepoint with the data available. If `inet_ntop` fails, it returns null and you get output like `nio_ch_Net_Connect(descriptor=13, connect_to(AF_INET6: addr=<NULL> port=33893 scope_id=0), length=28)`.

Reply:
That sounds reasonable.

Reply:
Also if `inet_ntop` fails I think it would be a bug and not a runtime problem.